src/main/antlr/org/hibernate/sqm/parser/hql/internal/antlr/HqlParser.g4

```diff
@@ -87,10 +88,26 @@ orderingSpecification
 	;
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// LIMIT/OFFSET clause
+
+limitClause
+	: LIMIT parameterOrNumberLiteral
+	;
+
+offsetClause
+	: OFFSET parameterOrNumberLiteral
+	;
+
+parameterOrNumberLiteral
+	: parameter
+	| INTEGER_LITERAL
+	;
```

Comment:
I'm more +1 for a warning than not being able to make use of parameters. As far as I know the most widespread DBs, all of them allow parameters. I'm fine with not caching in that case, but I would suggest that being configurable via a property.